src/lib/markbind/src/parser.js

```diff
@@ -18,888 +19,738 @@ _.pick = require('lodash/pick');
-}
-
-function createErrorNode(element, error) {
-  const errorElement = cheerio.parseHTML(utils.createErrorElement(error), true)[0];
-  return Object.assign(element, _.pick(errorElement, ['name', 'attribs', 'children']));
-}
-
-function createEmptyNode() {
-  const emptyElement = cheerio.parseHTML('<div></div>', true)[0];
-  return emptyElement;
-}
-
-function isText(element) {
-  return element.type === 'text' || element.type === 'comment';
-}
-
-function Parser(options) {
-  this._options = options || {};
-  // eslint-disable-next-line no-console
-  this._onError = this._options.errorHandler || console.error;
-  this._fileCache = {};
-  this.dynamicIncludeSrc = [];
-  this.staticIncludeSrc = [];
-  this.boilerplateIncludeSrc = [];
-  this.missingIncludeSrc = [];
-}
 
-Parser.resetVariables = function () {
-  VARIABLE_LOOKUP.clear();
-  FILE_ALIASES.clear();
-  PROCESSED_INNER_VARIABLES.clear();
-};
+  /**
+   * Returns an object containing the imported variables for specified file
+   * @param file file name to get the imported variables for
+   */
+  getImportedVariableMap(file) {
+    const innerVariables = {};
+    this.FILE_ALIASES.get(file).forEach((actualPath, alias) => {
+      innerVariables[alias] = {};
+      const variables = this.VARIABLE_LOOKUP.get(actualPath);
+      variables.forEach((value, name) => {
+        innerVariables[alias][name] = value;
+      });
+    });
+    return innerVariables;
+  }
 
-/**
- * Extract variables from an include element
- * @param includeElement include element to extract variables from
- * @param contextVariables variables defined by parent pages
- */
-function extractIncludeVariables(includeElement, contextVariables) {
-  const includedVariables = { ...contextVariables };
-  Object.keys(includeElement.attribs).forEach((attribute) => {
-    if (!attribute.startsWith('var-')) {
-      return;
-    }
-    const variableName = attribute.replace(/^var-/, '');
-    if (!includedVariables[variableName]) {
-      includedVariables[variableName] = includeElement.attribs[attribute];
-    }
-  });
-  if (includeElement.children) {
-    includeElement.children.forEach((child) => {
-      if (child.name !== 'variable' && child.name !== 'span') {
-        return;
-      }
-      const variableName = child.attribs.name || child.attribs.id;
+  /**
+   * Extract page variables from a page
+   * @param filename for error printing
+   * @param data to extract variables from
+   * @param userDefinedVariables global variables
+   * @param includedVariables variables from parent include
+   */
+  extractPageVariables(fileName, data, userDefinedVariables, includedVariables) {
+    const $ = cheerio.load(data);
+    const pageVariables = {};
+    this.VARIABLE_LOOKUP.set(fileName, new Map());
+    /**
+     * <import>ed variables have not been processed yet, we replace such variables with itself first.
+     */
+    const importedVariables = {};
+    const that = this;
```

Comment:
`const self = this;` is the more appropriate style that we use in the codebase. E.g.

https://github.com/MarkBind/markbind/blob/0acd97c3fd836ad49442f10e678b660d75924124/src/lib/markbind/src/parser.js#L314-L318